bin/echo.c

```diff
@@ -45,9 +45,11 @@ int negotiate(struct s2n_connection *conn)
     s2n_blocked_status blocked;
     do {
         if (s2n_negotiate(conn, &blocked) < 0) {
-            fprintf(stderr, "Failed to negotiate: '%s'. %s\n", s2n_strerror(s2n_errno, "EN"), s2n_strerror_debug(s2n_errno, "EN"));
-            fprintf(stderr, "Alert: %d\n", s2n_connection_get_alert(conn));
-            S2N_ERROR_PRESERVE_ERRNO();
+            if (s2n_error_get_type(s2n_errno) != S2N_ERR_T_BLOCKED) {
```

Comment:
Q: If an error is raised only during non-blocking mode, as in blocking mode we continue to block without raising an error, what is the check intended to do?

Reply:
The check returns an error if s2n_negotiate returns an error. If the error is related to blocking, we keep going.

I'm not sure I follow the question?

Reply:
That clears it up, S2N_ERR_T_BLOCKED is an error type that represents the blocked state and is not treated as an error.